docs/checkout-psp-api.yaml

```diff
@@ -460,6 +464,254 @@ paths:
+        '400':
+          description: Invalid list settlements request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /payments/report:
+    post:
+      tags:
+      - payment reports
+      summary: Request a payment report
+      description: |
+        Request a report of payments to be sent to the given callback url. Callback schema defined in PaymentReportCallbackJSON.
+      operationId: requestPaymentReport
+      parameters:
+        - name: checkout-account
+          in: header
+          description: Merchant ID
+          example: 375917
+          schema:
+            type: integer
+            format: int32
+        - name: checkout-algorithm
+          in: header
+          description: HMAC algorithm
+          example: sha512
+          schema:
+            type: string
+            enum:
+            - sha256
+            - sha512
+        - name: checkout-method
+          in: header
+          description: HTTP method of the request
+          example: POST
+          schema:
+            type: string
+            enum:
+            - GET
+            - POST
+        - name: checkout-timestamp
+          in: header
+          description: Current timestamp in ISO 8601 format
+          example: '2018-08-08T10:10:59Z'
+          schema:
+            type: string
+            format: date-time
+        - name: signature
+          in: header
+          description: HMAC signature calculated over the request headers and payload
+          schema:
+            type: string
+      requestBody:
+        description: Payment body payload
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PaymentReportRequest'
+      responses:
+        '200':
+          description: Payment report request response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PaymentReportRequest'
+        '400':
+          description: Invalid payment report request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /settlements/{settlementId}/payments/report:
+    post:
+      tags:
+      - payment reports
+      summary: Request a report of payments contained in a given settlement
+      description: |
+        Request a report of payments contained in a given settlement to be sent to the given callback url. Callback schema defined in PaymentReportCallbackJSON.
```

Comment:
Live, love, and learn and then do this: https://swagger.io/docs/specification/callbacks/